supervisor/api/backups.py

```diff
@@ -0,0 +1,220 @@
+import re
+from tempfile import TemporaryDirectory
+
+from aiohttp import web
+from aiohttp.hdrs import CONTENT_DISPOSITION
+import voluptuous as vol
+
+from ..backups.validate import ALL_FOLDERS
+from ..const import (
+    ATTR_ADDONS,
+    ATTR_BACKUPS,
+    ATTR_CONTENT,
+    ATTR_DATE,
+    ATTR_FOLDERS,
+    ATTR_HOMEASSISTANT,
+    ATTR_NAME,
+    ATTR_PASSWORD,
+    ATTR_PROTECTED,
+    ATTR_REPOSITORIES,
+    ATTR_SIZE,
+    ATTR_SLUG,
+    ATTR_SNAPSHOTS,
+    ATTR_TYPE,
+    ATTR_VERSION,
+    CONTENT_TYPE_TAR,
+)
+from ..coresys import CoreSysAttributes
+from ..exceptions import APIError
+from .utils import api_process, api_validate
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+RE_SLUGIFY_NAME = re.compile(r"[^A-Za-z0-9]+")
+
+# pylint: disable=no-value-for-parameter
+SCHEMA_RESTORE_PARTIAL = vol.Schema(
+    {
+        vol.Optional(ATTR_PASSWORD): vol.Any(None, vol.Coerce(str)),
+        vol.Optional(ATTR_HOMEASSISTANT): vol.Boolean(),
+        vol.Optional(ATTR_ADDONS): vol.All([vol.Coerce(str)], vol.Unique()),
+        vol.Optional(ATTR_FOLDERS): vol.All([vol.In(ALL_FOLDERS)], vol.Unique()),
+    }
+)
+
+SCHEMA_RESTORE_FULL = vol.Schema(
+    {vol.Optional(ATTR_PASSWORD): vol.Any(None, vol.Coerce(str))}
+)
+
+SCHEMA_BACKUP_FULL = vol.Schema(
+    {
+        vol.Optional(ATTR_NAME): vol.Coerce(str),
+        vol.Optional(ATTR_PASSWORD): vol.Any(None, vol.Coerce(str)),
+    }
+)
+
+SCHEMA_BACKUP_PARTIAL = SCHEMA_BACKUP_FULL.extend(
+    {
+        vol.Optional(ATTR_ADDONS): vol.All([vol.Coerce(str)], vol.Unique()),
+        vol.Optional(ATTR_FOLDERS): vol.All([vol.In(ALL_FOLDERS)], vol.Unique()),
+        vol.Optional(ATTR_HOMEASSISTANT): vol.Boolean(),
+    }
+)
+
+
+class APIBackups(CoreSysAttributes):
+    """Handle RESTful API for backups functions."""
+
+    def _extract_slug(self, request):
+        """Return backup, throw an exception if it doesn't exist."""
+        backup = self.sys_backups.get(request.match_info.get("slug"))
+        if not backup:
+            raise APIError("Backup does not exist")
+        return backup
+
+    @api_process
+    async def list(self, request):
+        """Return backup list."""
+        data_backups = []
+        for backup in self.sys_backups.list_backups:
+            data_backups.append(
+                {
+                    ATTR_SLUG: backup.slug,
+                    ATTR_NAME: backup.name,
+                    ATTR_DATE: backup.date,
+                    ATTR_TYPE: backup.sys_type,
+                    ATTR_PROTECTED: backup.protected,
+                    ATTR_CONTENT: {
+                        ATTR_HOMEASSISTANT: backup.homeassistant_version is not None,
+                        ATTR_ADDONS: backup.addon_list,
+                        ATTR_FOLDERS: backup.folders,
+                    },
+                }
+            )
+
+        if request.path == "/snapshots":
+            # Kept for backwards compability
+            return {ATTR_SNAPSHOTS: data_backups}
```

Comment:
Let's ditch that and we make a blog post and give software developers 2 weeks time to update their add-on. As we use the output on CLI, that confuse a lot to have all things twice

Reply:
I don't think 2 weeks is enough, the UI can't use the /backups endpoint until the user is on a core version after https://github.com/home-assistant/core/pull/51629 is released

Reply:
nevermind, that only applies to uploads, but I still think 2 weeks are too little

Reply:
Right, but we rename also the meta-file inside the backup. So one "big" breaking change. We can also wait 3 weeks after the blog post or 4 weeks, whatever you think makes sense

Reply:
We can never break loading `snapshot.json` (the metafile), doing that will render all previous backups useless